### DISCUSSION TITLE
Smart-Contract-Lottery: I have deployed the contract on testnet but I can't get a winner even though I have entered the raffle and upkeep is also returning true

### DISCUSSION THREAD

**Discussion by vabs10** [Cyfrin/foundry-full-course-cu#2470] [Q&A] [answered]
I had the deployed the contract for Lottery-Raffle on [Sepolia Testnet](https://sepolia.etherscan.io/address/0x5a7bf5dc85bba37f03c9941ed3d9b00c1636d479#readContract) it was verified so I went ahead and created a automation and VRF in chain link. After this I connected the wallet and entered the raffle and checked the upkeep's status which also changed to true but then I checked the VRF and it didn't get fullfilled. Can someone please help me understand what might be the issue ?

 Code : https://github.com/vabs10/foundry-smart-contract-lottery
 VRF : 
![image](https://github.com/user-attachments/assets/f51303a6-41ae-4c4e-bed0-16ce744469c8)
Automation: 
![image](https://github.com/user-attachments/assets/c491839d-77d9-4a68-bcd8-d1f9705d6ab5)
Smart-Contract-Lottery: https://sepolia.etherscan.io/address/0x5a7bf5dc85bba37f03c9941ed3d9b00c1636d479

**Comment by EngrPips** [comment]
Hello @vabs10, is this `0x5A7bf5DC85Bba37f03C9941ED3D9b00C1636D479` your lottery smart contract address?

**Reply by vabs10** [reply]
Hi @EngrPips, yes this is 0x5A7bf5DC85Bba37f03C9941ED3D9b00C1636D479 lottery smart contract address

**Reply by EngrPips** [reply]
It appears to me that you did not register the smart contract for chainlink automation. Am I wrong? The address shown on the Automation page you shared holds another contract that starts with `0xe2800` Did I miss something?

**Reply by vabs10** [reply]
Sorry for the misunderstanding this([0xE280086Ca52AECFfC1A5eBf7F3f1132B5A94b203](https://sepolia.etherscan.io/address/0xe280086ca52aecffc1a5ebf7f3f1132b5a94b203#readContract)) is also lottery smart contract address which I tried couple days ago and faced same issue.

![image](https://github.com/user-attachments/assets/31299c94-0554-417e-9b5f-72c31140f3f8)

![image](https://github.com/user-attachments/assets/3ac9fca1-745f-46cb-9bca-5ddb11cd7a38)

here are both the lottery smart contract I created

**Reply by EngrPips** [reply, marked as answer]
Hello @vabs10, can you raise the balance of your vrf subscription to 45 and retry?

**Reply by vabs10** [reply]
I have raised the balance of VRF subscription to 45, now should I enter the raffle again ?

**Reply by EngrPips** [reply]
I think you need to add this Lottery contract `0x5A7bf5DC85Bba37f03C9941ED3D9b00C1636D479` to your subscription for upkeep first and for this `[0xE280086Ca52AECFfC1A5eBf7F3f1132B5A94b203]` has it been up to 24 hours since you entered the lottery?

**Reply by vabs10** [reply]
For ```[0xE280086Ca52AECFfC1A5eBf7F3f1132B5A94b203]``` it's been more than 48 hours and after I created new upkeep for ```[0x5A7bf5DC85Bba37f03C9941ED3D9b00C1636D479]``` it got fulfilled.
![image](https://github.com/user-attachments/assets/82520c55-2d0e-460b-98dd-4d5ab7b1dad0)

![image](https://github.com/user-attachments/assets/ba602a68-9fbc-4a32-a4b1-a85239d2ced9)

Thanks, also why do I had to raise the balance to 45 ?

**Reply by EngrPips** [reply]
Honestly, I am not sure what is the motivation behind requiring people to have a high balance of Link on their vrf subscription before their request can be fulfill but that is what they require lol

**Reply by vabs10** [reply]
@EngrPips Thank you very much for solving this also that's just weird or waste of links.

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
